volumes/miovision/api/intersection_tmc.py

```diff
@@ -300,7 +299,7 @@ def insert_data(conn, start_time, end_iteration_time, table, dupes):
 
 def daterange(start_time, end_time, dt):
     """Generator for a sequence of regular time periods."""
-    for i in range(math.ceil((end_time - start_time) / dt) - 1):
+    for i in range(round((end_time - start_time) / dt)):
```

Comment:
Given we've already been burned by the math why not the below. And also, to make this easier to be copied to future work, only return one variable.

```python
 def daterange(start_time, end_time, delta): 
    i = 0
    while start_time + i * delta < end_time:
        yield start_time + i * delta
        i = i + 1
```

Reply:
This doesn't work in the case of a DST rollback, eg. when:

```python
start_time = local_tz.localize(dateutil.parser.parse(str("2020-11-01")))
end_time = local_tz.localize(dateutil.parser.parse(str("2020-11-02")))
```

in which case 2020-11-02 00:00 - 06:00 gets included. Rounding works for both cases of DST and normal times.